packages/fleet/lib/models/deployments.ts

```diff
@@ -0,0 +1,78 @@
+import type knex from 'knex';
+import type { Result } from '@nangohq/utils';
+import { Err, Ok, stringifyError } from '@nangohq/utils';
+import type { CommitHash, Deployment } from '../types.js';
+
+export const DEPLOYMENTS_TABLE = 'deployments';
+
+export interface DBDeployment {
+    readonly commit_id: CommitHash;
+    readonly created_at: Date;
+    readonly superseded_at: Date | null;
+}
+
+const DBDeployment = {
+    to(dbDeployment: DBDeployment): Deployment {
+        return {
+            commitId: dbDeployment.commit_id,
+            createdAt: dbDeployment.created_at,
+            supersededAt: dbDeployment.superseded_at
+        };
+    },
+    from(deployment: Deployment): DBDeployment {
+        return {
+            commit_id: deployment.commitId,
+            created_at: deployment.createdAt,
+            superseded_at: deployment.supersededAt
+        };
+    }
+};
+
+export async function create(db: knex.Knex, commitId: CommitHash): Promise<Result<Deployment>> {
+    try {
+        return db.transaction(async (trx) => {
+            const now = new Date();
+            // supersede any active deployments
+            await trx
+                .from<DBDeployment>(DEPLOYMENTS_TABLE)
+                .where({
+                    superseded_at: null
+                })
+                .update({ superseded_at: now });
+            // insert new deployment
+            const dbDeployment: DBDeployment = {
+                commit_id: commitId,
+                created_at: now,
+                superseded_at: null
+            };
+            const [inserted] = await trx.into<DBDeployment>(DEPLOYMENTS_TABLE).insert(dbDeployment).returning('*');
+            if (!inserted) {
+                return Err(new Error(`Error: no deployment '${commitId}' created`));
+            }
+            return Ok(DBDeployment.to(inserted));
+        });
```

Comment:
We don't really do it rn but there is no way to rollback a deploy? or maybe reinsert the same commitHash maybe?

Reply:
very good point. I forgot about rolling back. commitHash is indeed not unique. I will fix

Reply:
that's exactly what's this is doing. Deployments are immutable and rolling back means inserting a new entry with the same commitHash. No update, no deletion